### ISSUE TITLE
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 7082923680, available 7082732800)

### ISSUE THREAD

**Issue by imSrbh** [su77ungr/CASALIOY#3]
@su77ungr 
**I am having 32Cores and 64GB RAM.**

1. I am getting  `ggml_new_tensor_impl: not enough space in the context's memory pool (needed 7082923680, available 7082732800)
`

2. How we can restrict the token_length? and limit its domain to the ingested document file?

```
> Question:
who is saying that "save democracy"

 

> Answer:
The speaker is calling for the Senate to pass the Freedom to Vote Act, the John Lewis Voting Rights Act, and the Disclose Act to ensure that Americans have the right to vote and to know who is funding their elections.

 

> Time Taken: 39.02538466453552

 

Enter a query: what is the date today?

 

llama_print_timings:        load time =   227.71 ms
llama_print_timings:      sample time =     0.00 ms /     1 runs   (    0.00 ms per run)
llama_print_timings: prompt eval time =   334.69 ms /     7 tokens (   47.81 ms per token)
llama_print_timings:        eval time =     0.00 ms /     1 runs   (    0.00 ms per run)
llama_print_timings:       total time =   337.46 ms
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 7082923680, available 7082732800)
```

**Comment by imSrbh** [comment]
@su77ungr I am still getting this error..
```
Enter a query: what is vickey's email?

llama_print_timings:        load time =   200.62 ms
llama_print_timings:      sample time =     0.00 ms /     1 runs   (    0.00 ms per run)
llama_print_timings: prompt eval time =   429.41 ms /     8 tokens (   53.68 ms per token)
llama_print_timings:        eval time =   116.50 ms /     1 runs   (  116.50 ms per run)
llama_print_timings:       total time =   551.65 ms
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 6426660032, available 6418726608)
Segmentation fault (core dumped)
```

I am using latest changes only as you suggested about Llama module version.

**Comment by su77ungr** [comment]
That's your hardware bottlenecking. Try to ingest smaller files or upgrade.

**Comment by imSrbh** [comment]
I am using sample.csv only provided in the source_documents

**Comment by su77ungr** [comment]
Try building from docker. This is the first error I could not recreate. I'm running on 16GiB without a problem though. 

I'm searching for a fix

**Comment by hippalectryon-0** [comment]
@imSrbh  Does it still happen with the new release ?

**Comment by imSrbh** [comment]
@hippalectryon-0 

No, It's not with the new release.
The only issue I am facing is GUI:

there is no attribute in startLLM named initialize_qa_system but still it's been used in gui.

**Comment by hippalectryon-0** [comment]
Your new error is fixed in #58 , it's awaiting merge.

If your original issue is fixed, you can close the issue :)
